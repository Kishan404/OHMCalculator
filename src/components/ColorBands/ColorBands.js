import React, { Component } from 'react';
import ColorBand from '../ColorBand/ColorBand';
import PropTypes from 'prop-types';

const ColorBands =(props) => {
    return (
        <div className="colorbands">
            {props.bandsList.map((band,key)=>{
                return <ColorBand 
                            key ={key}
                            bandIndex = {key}
                            bandLabel = {band.bandLabel} 
                            bandOptions = {Object.entries(band.colorOptions)}
                            handleColorPick = {props.handleColorPick} 
                        />
            })}
        </div>
    );
}
ColorBands.propTypes = {
    bandsList : PropTypes.array,
    handleColorPick : PropTypes.func
}
export default ColorBands;