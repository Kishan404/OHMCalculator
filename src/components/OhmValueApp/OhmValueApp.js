import React, { Component } from 'react';
import PropTypes from 'prop-types';
import logo from '../../resources/resister.svg';
import './OhmApp.css';
import OhmValueCalculator from '../../OhmValueCalculator';
import ColorBands from '../ColorBands/ColorBands';


class OhmValueApp extends Component {
    constructor(props) {
        super(props);
        this.state = {
            firstDigitColor: props.firstDigitColor,
            secondDigitColor: props.secondDigitColor,
            multiplierColor: props.multiplierColor,
            toleranceColor: props.toleranceColor,
            resistance: props.resistance,
            tolerance: props.tolerance,
        };
        this.handleColorPick = this.handleColorPick.bind(this);
    }

componentDidMount() {

    let ohmcal = new OhmValueCalculator();
    let result = ohmcal.calculateOhmValue(this.props.firstDigitColor,this.props.secondDigitColor,this.props.multiplierColor,this.props.toleranceColor);
    console.log(result);
    this.setState(() => {
        return {
            resistance: result.Ohm,
            tolerance: result.tolerance,
        }
    });

}    
handleColorPick(newcolor,bandType) {

    this.setState(()=> {
        return {
            [bandType]: newcolor
        }
    });

    let ohmcal = new OhmValueCalculator();
    
    debugger;
    this.setState((prevState)=> {
        let result = ohmcal.calculateOhmValue(prevState.firstDigitColor,prevState.secondDigitColor,prevState.multiplierColor,prevState.toleranceColor);
       return {
            resistance: result.Ohm,
            tolerance: result.tolerance
        }
    });

  }


  render() {
    return (
      <div className="App">
        <img src={logo} className="App-logo" alt="logo" />
        <header className="App-header">
              
        </header>
        <h1 className="App-title">{this.props.hearder}</h1>
        
        <p className="App-intro">
          {this.props.subTitle}
          <br />
          Please select colors bands to calculate resitance value. Use the matching image pattern of your resister for correct input colors selection.  
        </p>
        <br/>
        <div>
          <div><ColorBands 
                bandsList = {this.props.bandsList}
                handleColorPick = {this.handleColorPick}
                /></div>
          <div></div>
        </div>

        <p><b>Resistance : {(this.state.resistance+' Ohm '+(this.state.tolerance ? 'Â± '+this.state.tolerance+'%':''))} </b></p>
        
      </div>
    );
  }
}

OhmValueApp.propTypes = {
    bandsList: PropTypes.array,
    hearder : PropTypes.string,
    subTitle : PropTypes.string,
    firstDigitColor: PropTypes.string,
    secondDigitColor: PropTypes.string,
    multiplierColor: PropTypes.string,
    toleranceColor: PropTypes.string,
};

export default OhmValueApp;
