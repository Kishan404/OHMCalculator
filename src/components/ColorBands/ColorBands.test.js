import React, { Component } from 'react';
const ReactTestRenderer = require('react-test-renderer');
import ColorBands from './ColorBands';
import OhmValueApp from '../OhmValueApp/OhmValueApp';

describe('Colorbands Snapshot test', () => {
    it('Should compare the component with a snapshot', () => {
      const component = ReactTestRenderer.create(
        <ColorBands
        bandsList = {[
        {
            bandLabel: "Color Band A",
            bandType: "firstDigitColor",
            colorOptions: {
                Brown: 1,
                Red: 2,
                Orange: 3,
                Yellow: 4,
                Green: 5,
                Blue: 6,
                Violet: 7,
                Gray: 8,
                White: 9
            }
        }]}
        handleColorPick = {OhmValueApp.handleColorPick} 
        />
      );
      const json = component.toJSON();
      expect(json).toMatchSnapshot();
    })
  });