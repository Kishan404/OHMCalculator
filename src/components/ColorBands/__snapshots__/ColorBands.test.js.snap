// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colorbands Snapshot test Should compare the component with a snapshot 1`] = `
<div
  className="colorbands"
>
  <div
    className="colorband"
  >
    <label
      className="label"
      htmlFor="band0"
    >
      Color Band A
    </label>
    <select
      className="colorselector brown"
      id="band0"
      onInput={[Function]}
    >
      <option
        className="coloroption brown"
        value="Brown"
      >
        Brown  (1)
      </option>
      <option
        className="coloroption red"
        value="Red"
      >
        Red  (2)
      </option>
      <option
        className="coloroption orange"
        value="Orange"
      >
        Orange  (3)
      </option>
      <option
        className="coloroption yellow"
        value="Yellow"
      >
        Yellow  (4)
      </option>
      <option
        className="coloroption green"
        value="Green"
      >
        Green  (5)
      </option>
      <option
        className="coloroption blue"
        value="Blue"
      >
        Blue  (6)
      </option>
      <option
        className="coloroption violet"
        value="Violet"
      >
        Violet  (7)
      </option>
      <option
        className="coloroption gray"
        value="Gray"
      >
        Gray  (8)
      </option>
      <option
        className="coloroption white"
        value="White"
      >
        White  (9)
      </option>
    </select>
  </div>
</div>
`;
